### PR TITLE
Add a demo of picking and moving object.

### PR DESCRIPTION
**Pull request by rongguodong**
Click to pick, click again to drop.
When an object is picked, it can be rotated with camera and moved along
radius by finger swipe.

GearVRf-DCO-1.0-Signed-off-by: Guodong Rong
rongguodong@hotmail.com

### CODE REVIEW

**Comment by rongguodong** [comment]
This is a demo for VR Hackathon.

Since GVRMeshEyePointee does not work well with Future<GVRMesh>, I temporarily use a blocking mesh loading with FutureWrapper to make it working. When GVRMeshEyePointee is modified to be compatible with Future<GVRMesh>, we can make this demo better with asynchronous mesh loading.

**Comment by thomasflynn** [comment]
ok, it runs and displays now.  though, it wasn't responding to touches on my screen.  didn't try in a headset yet.

**Comment by thomasflynn** [comment]
ok, Guodong showed me how it works.  i just wasn't using it right.  Chao, could you do an initial review of the code?  thanks.

File: GVRf/Framework/jni/gl/gl_texture.h [hunk 1]
```diff
@@ -32,7 +32,7 @@ class GLTexture {
         glBindTexture(target, id_);
         glTexParameteri(target, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
         glTexParameteri(target, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
-        glTexParameteri(target, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR);
+        glTexParameteri(target, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
```
Comment (JonShemitz) [inline GVRf/Framework/jni/gl/gl_texture.h]: This fixes the Solar System sample.

File: GVRf/Sample/gvr-pickandmove/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="org.gearvrf.pickandmove"
+    android:versionCode="1"
+    android:versionName="1.0" >
+
+    <uses-sdk
+        android:minSdkVersion="16"
+        android:targetSdkVersion="19" />
+    
+    <uses-permission android:name="android.permission.MOUNT_UNMOUNT_FILESYSTEMS"/>
```
Comment (JonShemitz) [inline GVRf/Sample/gvr-pickandmove/AndroidManifest.xml]: Not sure we need this (these?) permissions.

File: GVRf/Framework/jni/gl/gl_texture.h (hunk 1 above)
Comment (thomasflynn) [inline GVRf/Framework/jni/gl/gl_texture.h]: good.  though, why was i seeing a null mesh?  and why was it a random number of objects that were visible?

File: GVRf/Sample/gvr-pickandmove/res/values-sw600dp/dimens.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+
+
+<resources>
+
+    <!--
+         Customize dimensions originally defined in res/values/dimens.xml (such as
+         screen margins) for sw600dp devices (e.g. 7" tablets) here.
+    -->
+
```
Comment (JonShemitz) [inline GVRf/Sample/gvr-pickandmove/res/values-sw600dp/dimens.xml]: Do we really need this empty dimensions file?

File: GVRf/Framework/jni/gl/gl_texture.h (hunk 1 above)
Comment (JonShemitz) [inline GVRf/Framework/jni/gl/gl_texture.h]: > why was i seeing a null mesh? and why was it a random number of objects that were visible?

That's a really good question. In all my poking around today, I never saw a `null` mesh or a `null` texture, whether with the original `.notify()` code or @guodongrong 's  `.notifyAll()` change.

**Comment by chenchao1407** [comment]
OK. I am going to review it soon.

File: GVRf/Sample/gvr-pickandmove/.settings/org.eclipse.jdt.core.prefs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+eclipse.preferences.version=1
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.6
+org.eclipse.jdt.core.compiler.compliance=1.6
+org.eclipse.jdt.core.compiler.source=1.6
```
Comment (chenchao1407) [inline GVRf/Sample/gvr-pickandmove/.settings/org.eclipse.jdt.core.prefs]: I am not sure if it is necessary to merge.

Comment (JonShemitz) [inline GVRf/Sample/gvr-pickandmove/.settings/org.eclipse.jdt.core.prefs]: Yeah, probably shouldn't.

**Comment by chenchao1407** [comment]
Code seems ok to me. Runs well both with and without Samsung Gear. Can be merged after Guodong removes the .settings folder.

**Comment by rongguodong** [comment]
I have removed .setting and the unnecessary permissions. 

I also added head tracking indicator per Tom's request. Please review and test it again. Thanks!

**Comment by rongguodong** [comment]
Rebased with top of tree.

**Comment by chenchao1407** [comment]
I think good to go with this. Any objection?

**Comment by thomasflynn** [comment]
Runs, does what i expect.  Jon, i think he addressed your comments.  Any further comments?

**Comment by JonShemitz** [comment]
>  Any further comments?

Nope.

**Pull request merged**